### DISCUSSION TITLE
Uploading an Image to Pocketbase using Svelte's Form Actions

### DISCUSSION THREAD

**Discussion by DoormatIka** [pocketbase/pocketbase#2151] [Q&A] [answered]
I've been trying to use form actions for handling forms but I'm stuck on file uploads for Pocketbase. It throws a `"thumbnail: Missing required value."`. If I remove the Nonempty requirement, I get nothing posted on my database.

The form action:
```js
/** @type {import("@sveltejs/kit").Actions} */
export const actions = {
  add: async ({ request, locals }) => {
      const req = await request.formData();

      const img = req.get("img"); // this returns a File
      console.dir(img, { depth: null })
      await locals.pb // the pocketbase object
        .collection("subjects")
        .create({
            ...
            thumbnail: img
        })
    }
}
```

The HTML:
```html
<form enctype="multipart/form-data" method="post" action="?/add">
    <label>
        Image
        <input type="file" name="img" accept="image/jpeg,image/png" required />
    </label>
</form>
```

What I get when I try to upload an image:
![image](https://user-images.githubusercontent.com/68234036/227589201-0e0b2e4d-98d4-4dbb-8f21-7dad58763192.png)

I have the parameter set up like this:
![image](https://user-images.githubusercontent.com/68234036/227588071-35651ef7-8e55-43d3-b92d-c68246edc5fa.png)

**Comment by ganigeorgiev** [comment, marked as answer]
You need to provide a [`FormData` instance](https://developer.mozilla.org/en-US/docs/Web/API/FormData) in order the underlying fetch call to detect the request as `multipart/form-data`.

Or in other words, if `img` in the above is a `File` or `Blob` instance, then you can do something like:

```js
const data = new FormData();
data.append("thumbnail", img);
await locals.pb.collection("subjects").create(data)
```

Alternatively, you can also pass directly the SvelteKit `FormData` object if you use the same name for your input as the field name (`thumbnail`).

If you haven't already, you can also find a plain JS file upload example in https://pocketbase.io/docs/files-handling/#uploading-files.

**Reply by DoormatIka** [reply]
Where exactly do I provide the FormData instance?

**Reply by ganigeorgiev** [reply]
It's the `data` variable in the above.

The SvelteKit's `req` in your example is also a `FormData` instance. If you name your input `thumbnail` then you should be able to pass it directly to the `create` call.
